src/main/scala/ignition/core/http/AsyncHttpClientStreamApi.scala

```diff
@@ -1,17 +1,35 @@
 package ignition.core.http
 
 import java.io.InputStream
+import java.net.{URL, URLDecoder, URLEncoder}
 import java.util.concurrent.TimeUnit
 
 import akka.util.Timeout
+import spray.http.Uri.{Path, Query}
 import spray.http._
 
 import scala.concurrent.Future
 import scala.concurrent.duration._
 import scala.language.postfixOps
 
-
 object AsyncHttpClientStreamApi {
+
+  private def sanitizePathSegment(segment: String) =
+    URLEncoder.encode(URLDecoder.decode(segment, "UTF-8"), "UTF-8").replace("+", "%20")
```

Comment:
@fernandors87 ok, this is reasonable. Just put a commentary like:
```scala
// Due to ancient standards, Java will encode space as + instead of using percent. See http://stackoverflow.com/questions/1634271/url-encoding-the-space-character-or-20 and https://docs.oracle.com/javase/7/docs/api/java/net/URLEncoder.html#encode(java.lang.String,%20java.lang.String)
```